app/Nova/Actions/DistributeShirt.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Nova\Actions;
+
+use Illuminate\Bus\Queueable;
+use Laravel\Nova\Actions\Action;
+use Illuminate\Support\Collection;
+use Laravel\Nova\Fields\ActionFields;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+
+class DistributeShirt extends Action
+{
+    use InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Perform the action on the given models.
+     *
+     * @param  \Laravel\Nova\Fields\ActionFields  $fields
+     * @param  \Illuminate\Support\Collection  $models
+     * @return mixed
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        $failures = [];
+        foreach ($models as $model) {
+            if ($model->package()->first()->eligible_for_shirt) {
+                if ($model->is_paid) {
+                    if (null === $model->swag_shirt_provided) {
+                        $model->swag_shirt_provided = date('Y-m-d H:i:s');
+                        $model->swag_shirt_providedBy = \Auth::user()->id;
+                        $model->save();
+                    } else {
+                        $this->markAsFailed($model, null);
+                        if (count($models) === 1) {
+                            return Action::danger('T-shirt already picked up.');
+                        } else {
+                            $failures[] = $model->id;
+                        }
+                    }
+                } else {
+                    $this->markAsFailed($model, null);
+                    if (count($models) === 1) {
+                        return Action::danger('This transaction is not yet paid.');
```

Comment:
As discussed in Slack, see if we can hide the action if the txn isn't paid.

Reply:
Done